src/main/java/org/apache/maven/shared/verifier/Verifier.java

```diff
@@ -115,61 +115,157 @@
 
     private static MavenLauncher embeddedLauncher;
 
+    private String settingsFile;
+
+    private boolean debug;
+
     public Verifier( String basedir )
         throws VerificationException
     {
-        this( basedir, null );
+        this.basedir = basedir;
+
+        this.forkMode = System.getProperty( "verifier.forkMode" );
+
+        findLocalRepo( settingsFile );
+
+        this.mavenHome = System.getProperty( "maven.home" );
+
+        useWrapper = Files.exists( Paths.get( getBasedir(), "mvnw" ) );
+
+        this.defaultCliArguments = DEFAULT_CLI_ARGUMENTS.clone();
+
     }
 
+    /**
+    *
+    * @deprecated to be removed
+    * use {@link #Verifier(String basedir)}
+     *
+    * */
+    @Deprecated
     public Verifier( String basedir, boolean debug )
         throws VerificationException
     {
         this( basedir, null, debug );
```

Comment:
I would to change body of constructors to show replacements, so should be:
```
  this(basedir);
 ... 
 needed set methods
```
in all deprecated constructor

Reply:
Okay, I get your point. I will update accordingly.